Dfc.CourseDirectory/Resources/ArmTemplates/template.json

```diff
@@ -26,6 +26,18 @@
         "description": "API host key (code) needed to access the postcode search"
       }
     },
+    "getAddressApiUrl": {
+      "type": "string",
+      "metadata": {
+        "description": "API URL for GetAddress postcode search"
```

Comment:
Does this allow a c# format placeholder for postcode? If so this might be a good place to document the expected format.

Reply:
A pointless exercise I expect. Someone is never going to refer to this a source of documentation having not already looked at the self documenting code.

Reply:
> "A pointless exercise I expect. "

I'm not sure I appreciate your tone.

> "Someone is never going to refer to this a source of documentation having not already looked at the self documenting code."

This is specifically documentation for people configuring the service who may not be the current team and may very well not want to have to read all the source code to find out that there's a `string.format` in there somewhere; if they even have the knowledge of c# to do so.

Personally out of empathy for future self/others I would document that it accepts `https://..../.../{postcode}/...`, if I understand the code correctly. Whether you bother is your call, it's just a suggestion.

Reply:
On reflection that's a fair point, will update.

Reply:
> I'm not sure I appreciate your tone.

If nobody ever looks at it again, it's a pointless exercise.

> if they even have the knowledge of c# to do so

They'll need to know about the C# code to understand the string formatting, so the comment is useless without the context of the code.

Reply:
> If nobody ever looks at it again, it's a pointless exercise.

consider it feedback about the way that came across in response to my efforts to improve our contributions to the codebase. "pointless" is pejorative.
 
> They'll need to know about the C# code to understand the string formatting, so the comment is useless without the context of the code.

I think a simple placeholder replacement could be understood without knowing all of c#, but that aside if you don't feel it's worth the effort to alter this that is your call. Feel free to resolve the comment and move on